zcutil/make-release.py

```diff
@@ -330,6 +349,37 @@ def sh_log(*args):
         raise SystemExit('Nonzero exit status: {!r}'.format(status))
 
 
+def sh_progress(markers, *args):
+    try:
+        import progressbar
+    except:
+        sh_log(*args)
+        return
+
+    PIPE = subprocess.PIPE
+    try:
+        p = subprocess.Popen(args, stdout=PIPE, stderr=PIPE, stdin=None)
```

Comment:
**tl;dr:** Change `stderr=PIPE` to `stderr=subprocess.STDOUT` here and in `sh_log`.

I believe I have seen a buffering problem that can lead to deadlock when using `stdout=PIPE, stderr=PIPE`, and my mental model goes something like this: if the producer (child process) writes a bunch to one stream while the consumer (parent process) is blocked on reading the other stream, then a deadlock can occur.

Also, I don't see `p.stderr` being read anywhere at all, which would have the effect of silencing it. I just noticed that `sh_log` has the same bug. I think the best fix in both cases is to change to `stderr=subprocess.STDOUT`. Would that break the progress bar feature? In my opinion it should do the right thing for `sh_log`.

Reply:
I tested this (by introducing a bug into the build), and it doesn't break the progress bar. The progress jumps to 100% (detecting the end of the make process) and then halts because the process status is non-zero.